snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -ex
+
+security-secretstore-setup -confdir "$SNAP_DATA/config/security-secretstore-setup/res" --vaultInterval=10
+
+# if redis pw file doesn't exist, handle initialization & read it
+if [ ! -f "$REDIS5_PASSWORD_PATHNAME" ]; then
```

Comment:
I'm still not clear on when this script is called when there is not a password defined?

Reply:
This script gets called once, on first boot when the redis password file doesn't exist. This is a workaround to the fact that redis can't read the password directly from Vault.